### PR TITLE
fix(auth-server): misplaced text in emails

### PR DESCRIPTION
**Pull request by xlisachan**
Because:
* “Starting with your next bill, your charge will change from $9.99 per year to $59.88” should include the previous payment cycle and new payment cycle in cases where a user moves from a monthly subscription to an annual subscription. It should look similar to “Starting with your next bill, your charge will change from $9.99 per month to $59.88 per year”

This commit:
* Fixes the issue in the `subscriptionUpgrade` subscription platform email as well as `subscriptionDowngrade`

Closes: #11374

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)
Upgrade - old template
Screen Shot 2021-12-21 at 10 37 46 AM

Upgrade - new template
Screen Shot 2021-12-22 at 1 49 35 PM

Downgrade - old template
Screen Shot 2021-12-21 at 10 37 55 AM

Downgrade - new template
Screen Shot 2021-12-22 at 1 49 24 PM

Please attach the screenshots of the changes made in case of change in user interface.

## Other information (Optional)
While the issue only addressed updates to the `subscriptionUpgrade` email, revisions have also been included for the `subscriptionDowngrade` email as well.

### CODE REVIEW

**Review by jodyheavener — DISMISSED** [review]
🤠

**Comment by xlisachan** [comment]
> 🤠

Lol @jodyheavener! There may actually be further changes to this.

File: packages/fxa-auth-server/lib/senders/emails/templates/subscriptionDowngrade/en.ftl [hunk 1]
```diff
@@ -10,7 +10,7 @@ subscriptionDowngrade-content-switch = You have successfully switched from { $pr
 # $paymentAmountNew (String) - The amount of the new subscription payment, including currency, e.g. $10.00
 # $productPaymentCycle (String) - The interval of time from the end of one payment statement date to the next payment statement date, e.g. month
 # $paymentProrated (String) - The one time fee to reflect the higher charge for the remainder of the payment cycle, including currency, e.g. $10.00
-subscriptionDowngrade-content-charge = Starting with your next bill, your charge will change from { $paymentAmountOld } per { $productPaymentCycle } to { $paymentAmountNew }. At that time you will also be given a one-time credit of { $paymentProrated } to reflect the lower charge for the remainder of this { $productPaymentCycle }.
+subscriptionDowngrade-content-charge = Starting with your next bill, your charge will change from { $paymentAmountOld } to { $paymentAmountNew } per { $productPaymentCycle }. At that time you will also be given a one-time credit of { $paymentProrated } to reflect the lower charge for the remainder of this { $productPaymentCycle }.
```
Comment (LZoog) [inline packages/fxa-auth-server/lib/senders/emails/templates/subscriptionDowngrade/en.ftl:13]: Darn, [my original comment in your other PR](https://github.com/mozilla/fxa/pull/11410#discussion_r773507992) applies here. We have to update the string IDs.

**Review by LZoog — CHANGES_REQUESTED** [review]
@xlisachan this actually affects the old templates as well, since this was found in production.

We need to change the strings in the old template HTML files and tests.

File: packages/fxa-auth-server/lib/payments/stripe.ts [hunk 2]
```diff
@@ -2196,7 +2196,7 @@ export class StripeHelper {
 
     const abbrevProductNew = await this.expandAbbrevProductForPlan(planNew);
     const {
-      interval: productPaymentCycle,
+      interval: productPaymentCycleNew,
```
Comment (xlisachan) [inline packages/fxa-auth-server/lib/payments/stripe.ts:2199]: I'm not sure if I needed to change this.

Comment (xlisachan) [reply packages/fxa-auth-server/lib/payments/stripe.ts:2199]: @biancadanforth @chenba, I am not sure if I need to update `productPaymentCycle` in `stripe.ts`, can you please review and let me know if this should be included in the fix? Thank you! 🙂

Comment (chenba) [reply packages/fxa-auth-server/lib/payments/stripe.ts:2199]: The renaming make sense.  But I cannot find where `productPaymentCycleOld` for the template is coming from.

Comment (xlisachan) [reply packages/fxa-auth-server/lib/payments/stripe.ts:2199]: Thanks for your feedback, @chenba! I have updated the PR to include `productPaymentCycleOld` for the template. When you have a moment, can you please take another look? Thanks again!

File: packages/fxa-auth-server/lib/payments/stripe.ts [hunk 3]
```diff
@@ -2422,6 +2422,7 @@ export class StripeHelper {
     const {
       amount: paymentAmountOldInCents,
       currency: paymentAmountOldCurrency,
+      interval: productPaymentCycleOld,
```
Comment (xlisachan) [inline packages/fxa-auth-server/lib/payments/stripe.ts:2425]: @chenba , I believe `productPaymentCycleOld` should be included here. Please review when you have a moment. Thank you!

**Comment by chenba** [comment]
@xlisachan @LZoog how does QA verify a PR that include changes to both old and new templates?  IIUC only old _or_ new templates can be enabled on stage?

**Comment by LZoog** [comment]
@chenba right now on staging, the new templates are enabled for `@mozilla.com` emails. Any others will receive the old templates on staging.

**Review by chenba — APPROVED** [review]
📧

**Comment by chenba** [comment]
> @chenba right now on staging, the new templates are enabled for `@mozilla.com` email addresses. Any others will receive the old templates on staging.

I don't like it...I love it.

**Comment by LZoog** [comment]
@chenba I didn't dig in here but I'm curious where `productPaymentCycleOld` is pulled in from the Stripe API response? I haven't worked with the emails with Stripe data much so maybe it's all good, just not sure where [that value would come from](https://stripe.com/docs/api/plans/object) 🤔

**Comment by chenba** [comment]
> @chenba I didn't dig in here but I'm curious where `productPaymentCycleOld` is pulled in from the Stripe API response? I haven't worked with the emails with Stripe data much so maybe it's all good, just not sure where [that value would come from](https://stripe.com/docs/api/plans/object) 🤔

The subscription update emails are triggered by webhooks.   The old plan can be found in the `previous_attributes` of the webhook event data.

**Review by LZoog — APPROVED** [review]
😎

**Pull request merged**